src/hypergraph.jl

```diff
@@ -14,35 +14,61 @@ A hypergraph storing information about vertices and hyperedges.
 **Constructors**
 
     Hypergraph{T}(n,k) where {T<:Real}
+    Hypergraph{T,V}(n, k) where {T<:Real, V}
+    Hypergraph{T,V,E}(n, k) where {T<:Real, V, E}
 
 Construct a hypergraph with a given number of vertices and hyperedges.
+Optionally, values of type `V` can be stored at vertices and values of type `E`
+can be stored at hyperedges.
 
-    Hypergraph(m::AbstractMatrix{T}) where {T<:Real}
+    Hypergraph(m::AbstractMatrix{Union{T, Nothing}}) where {T<:Real}
+    Hypergraph{V}(m::AbstractMatrix{Union{T, Nothing}}) where {T<:Real, V}
+    Hypergraph{V, E}(m::AbstractMatrix{Union{T, Nothing}}) where {T<:Real, V, E}
 
 Construct a hypergraph using its matrix representation.
 In the matrix representation rows are vertices and columns are hyperedges.
+Optionally, values of type `V` can be stored at vertices and values of type `E`
+can be stored at hyperedges.
 
 **Arguments**
 
-* `T` : type of values stored in the hypergraph
+* `T` : type of weight values stored in the hypergraph
+* `V` : type of values stored in the vertices of the hypergraph
+* `E` : type of values stored in the edges of the hypergraph
 * `n` : number of vertices
 * `k` : number of hyperedges
 * `m` : a matrix representation rows are vertices and columns are hyperedges
 
 """
-struct Hypergraph{T} <: AbstractMatrix{Union{T, Nothing}}
+struct Hypergraph{T,V,E} <: AbstractMatrix{Union{T, Nothing}}
     v2he::Vector{Dict{Int,T}}
     he2v::Vector{Dict{Int,T}}
-    Hypergraph{T}(n, k) where {T<:Real} =
-        new([Dict{Int,T}() for i in 1:n], [Dict{Int,T}() for i in 1:k])
+    vs::Vector{V}
```

Comment:
maybe beter `v_meta`  and `he_meta`? Not a crucial thing.

Reply:
One argument for `vs` is shorter code. I am not sure whether `_meta` is going to increase readibility. 
@spagnuolocarmine what do you think (since you proposed issue #2)

Reply:
I do not want to stick with `_meta`, but for me when I see `vs` it is like variables `k`, `l`, `m` in old Fortran code - I just do not know what they mean when I see them.